### PR TITLE
PLT-1091: Delete Ember Models on Sign Out

### PR DESCRIPTION
**Pull request by mbowen**

### CODE REVIEW

File: frontend/tests/acceptance/session-state-test.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  });
+
+  andThen(function() {
+    click('#sign-in');
+  });
+
+  andThen(function() {
+    assert.equal(currentURL(), '/account/edit-profile');
+    assert.equal(find('input#display_name').val(), 'jdoe0');
+  });
```
Comment (jonocodes) [inline frontend/tests/acceptance/session-state-test.js]: can you check against jdoe* instead. then the number does not matter. also do this for the final test.

File: frontend/tests/acceptance/session-state-test.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    Ember.run(application, 'destroy');
+  }
+});
+
+test('sign in, sign out', function(assert) {
+
+  visit('/session/login');
+
+  andThen(function() {
+    fillIn('input#email_address', 'aturing0@princeton.edu');
```
Comment (jonocodes) [inline frontend/tests/acceptance/session-state-test.js]: set this value to 'valid@email.com' so it does not imply the value is actually used in the test.

File: frontend/tests/acceptance/session-state-test.js (hunk 1 above)
Comment (mbowen) [inline frontend/tests/acceptance/session-state-test.js]: i'm sure we can do this, but i don't know how. i experimented with server.create supplying overrides for display name flavors, but was unable to hook this into when create is called under-the-covers. our ember tests are complicated by fact that a number of different approaches have been taken with no consolidation (ex: mirage data vs factory guy data).

Comment (mbowen) [inline frontend/tests/acceptance/session-state-test.js]: that said, i did push change for your other suggestion.

**Pull request merged**